addons/website_blog/static/tests/tours/blog_menu_test.js

```diff
@@ -0,0 +1,27 @@
+/** @odoo-module **/
+
+import tour from "web_tour.tour";
+
+tour.register("blog_menu_active_element", {
+        url: "/",
+        test: true,
+}, [
+    {
+        content: "Click on the first blog post link in the navbar",
+        trigger: "nav a[href^='/blog/']:first",
+    },
+    {
+        content: "Check if the first blog post link has the active class",
+        trigger: "a[href^='/blog/']:first.active",
+        run: () => {}, // it's a check
+    },
+    {
+        content: "Click on the second blog post link in the navbar",
+        trigger: "a[href^='/blog/']:eq(1)",
+    },
+    {
+        content: "Check if the second blog post link has the active class",
```

Comment:
Isn't there multiple workers on runbot?
Isn't the issue not systematic since you could reach another worker after changing page?
Maybe that step would need to be done in a loop 10 times?

The best solution would have to be able to test it in python only but it's not possible AFAIK since we wouldn't be able to provide a `httprequest.url` for `_is_active()` to work as if it was a client browser call.
Maybe with a mock tho.. Maybe it's worth investigating regular mocks or `MockRequest` (maybe just need to add "url" in the mockrequest) and do a layout rendering in python and checking the result (checking active is on the correct element with etree).

It would avoid any worker issue, if there is any to start with?

Reply:
> Isn't there multiple workers on runbot?

AFAICS, the tests are not done with multiple workers.
https://runbot121.odoo.com/runbot/static/build/59763640-master/logs/test_only.txt

I will check to do it only with pyhton :eyes:

Reply:
Ok @rdeodoo the test is now fully done in Python :ok_hand: